internal/repositories/app.go

```diff
@@ -1,36 +1,31 @@
 package repositories
 
 import (
+	"database/sql"
+	"github.com/integrity-sum/internal/core/models"
+	"github.com/integrity-sum/pkg/api"
 	"github.com/sirupsen/logrus"
-
-	"github.com/integrity-sum/internal/core/ports"
 )
 
 type AppRepository struct {
-	ports.IHashRepository
 	logger *logrus.Logger
+	db     *sql.DB
 }
 
-func NewAppRepository(logger *logrus.Logger) *AppRepository {
+func NewAppRepository(logger *logrus.Logger, db *sql.DB) *AppRepository {
 	return &AppRepository{
-		IHashRepository: NewHashRepository(logger),
-		logger:          logger,
+		logger: logger,
+		db:     db,
 	}
 }
 
 // IsExistDeploymentNameInDB checks if the base is empty
 func (ar AppRepository) IsExistDeploymentNameInDB(deploymentName string) (bool, error) {
-	db, err := ConnectionToDB(ar.logger)
-	if err != nil {
-		ar.logger.Error("failed to connection to database %s", err)
-		return false, err
-	}
-	defer db.Close()
 
 	var count int
 	query := "SELECT COUNT(*) FROM hashfiles WHERE name_deployment=$1 LIMIT 1;"
-	row := db.QueryRow(query, deploymentName)
-	err = row.Scan(&count)
+	row := ar.db.QueryRow(query, deploymentName)
+	err := row.Scan(&count)
```

Comment:
TODO: simplify - return true on condition

Reply:
Done

Reply:
Not exactly.. You've done almost the same, but i mean something like this:
```go
func RowExists(condition string) (rez bool) {
  // general, for any search:
	query := fmt.Sprintf("SELECT TRUE FROM hashfiles WHERE %s", condition)
	_ = db.QueryRow(query).Scan(&rez)
	return // returns true only if found rows that match condition

// in our case we may pass just a name to search, and refactor this to func iSDeploymentExists(name string)
}
...
if RowExists(name_deployment='any_name') {...
```
It's not really important, so leaving it up to you